src/main/java/com/github/dockerjava/core/command/CopyArchiveToContainerCmdImpl.java

```diff
@@ -131,11 +125,18 @@ public Void exec() throws NotFoundException {
                         "Only one of host resource or tar input stream should be defined to perform the copy, not both");
             }
             // We compress the given path, call exec so that the stream is consumed and then close it our self
-            try (InputStream uploadStream = buildUploadStream(this.hostResource, this.dirChildrenOnly)) {
-                this.tarInputStream = uploadStream;
+            Path toUpload = null;
+            try {
+                toUpload = Files.createTempFile("docker-java", ".tar.gz");
+                CompressArchiveUtil.tar(Paths.get(hostResource), toUpload, true, dirChildrenOnly);
+                this.tarInputStream = Files.newInputStream(toUpload);
```

Comment:
should this be wrapped with try-with-resources or it closed in somewhere else place?